src/coffea/nanoevents/transforms.py

```diff
@@ -487,6 +487,58 @@ def eventindex(stack):
     stack.append(out)
 
 
+def zeros_from_content_form(source_form):
+    form = copy.deepcopy(source_form)
+    form["form_key"] = concat(source_form["form_key"], "!zeros_from_content")
+    if not (form["class"] == "NumpyArray" or form["class"].startswith("ListOffset")):
+        raise RuntimeError
+    if form["class"].startswith("ListOffset"):
+        form["content"]["form_key"] = concat(
+            source_form["form_key"], "!zeros_from_content", "!content"
+        )
+    return form
+
+
+def zeros_from_content(stack):
+    source = stack.pop()
+    stack.append(awkward.zeros_like(source))
+
+
+def zeros_from_offsets_form(offsets_form):
+    if not offsets_form["class"].startswith("NumpyArray"):
+        raise RuntimeError
+
+    form = {
+        "class": "ListOffsetArray",
+        "offsets": "i64",
+        "content": {
+            "class": "NumpyArray",
+            "primitive": "float32",
+            "itemsize": 8,
+            "format": "i",
+            "form_key": concat(
+                offsets_form["form_key"], "!zeros_from_offsets", "!content"
+            ),
+        },
+        "form_key": concat(offsets_form["form_key"], "!zeros_from_offsets"),
+        "parameters": offsets_form.get("parameters", None),
```

Comment:
Is this going to copy the `__doc__` parameter from offsets? This might not be desirable.